src/main/kotlin/com/msd/robot/application/RobotApplicationService.kt

```diff
@@ -314,21 +280,30 @@ class RobotApplicationService(
     }
 
     /**
-     * Makes the specified [Robot] use the specified [ReparationItem][RepairItemType].
+     * Execute the list of RepairItemUsageCommands by:
+     *   - Making the specified [Robot] use the specified [ReparationItem][RepairItemType].
+     *   - Handling failures by calling the eventSender without stopping the execution of the remaining commands.
+     *   - Sending events after successful usage.
      *
-     * @param command the [RepairItemUsageCommand] which specifies which `Robot` should use which item
+     * @param commands a list of [RepairItemUsageCommand]s which specify which `Robot` should use which item
      */
-    fun useRepairItem(command: RepairItemUsageCommand) {
-        val robots = robotDomainService.useRepairItem(command.robotUUID, command.itemType)
-        eventSender.sendEvent(
-            ItemRepairEventDTO(
-                true,
-                "Robot has used ${command.itemType}",
-                robots
-            ),
-            EventType.ITEM_REPAIR,
-            command.transactionUUID
-        )
+    fun executeRepairItemUsageCommands(commands: List<RepairItemUsageCommand>) {
+        commands.forEach { command ->
+            try {
+                val robots = robotDomainService.useRepairItem(command.robotUUID, command.itemType)
+                eventSender.sendEvent(
```

Comment:
Why did you only extract some of the event sending to `SuccessEventSender` and not all?